### PR TITLE
#2559. Add constructor augmentation tests. Part 3.

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+/// parameter scope that the augmenting body is executing in.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A10_t03.dart';
+
+augment class C {
+  augment C(int x, [int y = 0]) {
+    _log += "C($x, $y);";
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart:24]: The really weird thing is that if we do `y++` just before `augmented()` then the augmented constructor body should have a `y` whose value is the same as the value of the parameter `y` in this body. So `y++` here should determine the outcome of an evaluation of `y` in the augmented constructor's body, not just the outcome of an evaluation of `y` in this body.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - In the augmenting constructor's body, an `augmented()` call executes the
+///     augmented constructor's body in the same parameter scope that the
+///     augmenting body is executing in. The expression has type `void` and
+///     evaluates to `null`.
+///
+/// @description Checks that it is a compile-time error to use value returned by
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart:13]: Typo:
```suggestion
/// @description Checks that it is a compile-time error to use the value returned by
```

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+/// `augmented()` call in an augmenting constructor body.
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart:14]: Perhaps:
```suggestion
/// the invocation of `augmented()` in an augmenting constructor body.
```

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06_lib.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - In the augmenting constructor's body, an `augmented()` call executes the
+///     augmented constructor's body in the same parameter scope that the
+///     augmenting body is executing in. The expression has type `void` and
+///     evaluates to `null`.
+///
+/// @description Checks that it is a compile-time error to use value returned by
+/// `augmented()` call in an augmenting constructor body.
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06_lib.dart:14]: Similar.

**Review by eernstg — COMMENTED** [review]
Looks good! I think it would be helpful to adjust a couple of comments.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A09_t03.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+///   - If the augmenting constructor has an explicit block body, then that body
+///     replaces any existing constructor body.
+///
+/// @description Checks that it is a compile-time error to augment a default
+/// unnamed constructor.
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A09_t03.dart]: Added

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A09_t03_lib.dart [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+///   - If the augmenting constructor has an explicit block body, then that body
+///     replaces any existing constructor body.
+///
+/// @description Checks that it is a compile-time error to augment a default
+/// unnamed constructor.
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A09_t03_lib.dart]: Added

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart:24]: `x++` added

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart (hunk 2 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart]: Fixed

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart (hunk 3 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06.dart]: Updated

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06_lib.dart (hunk 4 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t06_lib.dart]: Updated

**Review by sgrekhov — COMMENTED** [review]
Thank you! Updated. PTAL.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart (hunk 1 above)
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart:24]: Thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**